src/pages/TermsOfService.test.js

```diff
@@ -0,0 +1,65 @@
+import { render, screen } from '@testing-library/react'
+import { act } from 'react-dom/test-utils'
+import { h } from 'react-hyperscript-helpers'
+import { Ajax } from 'src/libs/ajax'
+import { authStore } from 'src/libs/state'
+import TermsOfServicePage from 'src/pages/TermsOfService'
+
+
+jest.mock('src/libs/ajax')
+jest.mock('src/libs/notifications')
+
+const setupMockAjax = termsOfService => {
+  const getTos = jest.fn().mockReturnValue(Promise.resolve('some text'))
+  const getTermsOfServiceAdherenceStatus = jest.fn().mockReturnValue(termsOfService)
+  const getStatus = jest.fn().mockReturnValue({})
+  Ajax.mockImplementation(() => ({
+    Metrics: {
+      captureEvent: jest.fn()
+    },
+    User: {
+      profile: {
+        get: jest.fn().mockReturnValue(Promise.resolve({ keyValuePairs: [] })),
+      },
+      getTos,
+      getTermsOfServiceAdherenceStatus,
+      getStatus
+    }
+  }))
+}
+
+describe('TermsOfService', () => {
+  afterEach(() => {
+    authStore.reset()
+  })
+
+  it('shows "Continue under grace period" when the user has not accepted the latest ToS but is still allowed to use Terra', async () => {
+    const termsOfService = {
+      userHasAcceptedLatestTos: false,
+      permitsSystemUsage: true,
+    }
+    const isSignedIn = true
+
+    setupMockAjax(termsOfService)
+
+    render(h(TermsOfServicePage))
+    authStore.update(state => ({ ...state, termsOfService, isSignedIn }))
+    const continueUnderGracePeriodButton = await screen.findByText('Continue under grace period')
+    expect(continueUnderGracePeriodButton).not.toBeFalsy()
+  })
+
+  it('does not show "Continue under grace period" when the user has not accepted the latest ToS and is not allowed to use Terra', async () => {
+    const termsOfService = {
+      userHasAcceptedLatestTos: false,
+      permitsSystemUsage: false,
+    }
+    const isSignedIn = true
+    setupMockAjax(termsOfService)
+
+    // Need to wrap in 'act' or else get a warning about updating react state
+    await act(() => Promise.resolve(render(h(TermsOfServicePage))))
+    authStore.update(state => ({ ...state, termsOfService, isSignedIn }))
+    const continueUnderGracePeriodButton = await screen.queryByText('Continue under grace period')
+    expect(continueUnderGracePeriodButton).not.toBeInTheDocument()
+  })
+})
```

Comment:
I think it might be worth adding a few more test cases like when stuff is undefined. Are there any cases in that chart kai and greg made we can unit test?

Reply:
I've boiled down what the UI expects into two variables. 

* The user hasn't accepted latest and can use Terra: show the notification and "continue under grace period button"
* The user hasn't accepted the latest and cannot use Terra: Redirect to ToS page and force acceptance
* The user has accepted latest and can use Terra: No change

The "User has accepted latest and cannot use Terra" is an impossibility, invalid state.

Reply:
So, I think 1 more test case is probably warranted.